Imports.md

```diff
@@ -0,0 +1,407 @@
+# ![wgsl-import-spec](branding.svg)
+
+# Summary
+
+We propose adding an importing mechanism to the WGSL shading language as an extension.
+
+# Motivation
+
+When writing bigger WGSL shaders, one tends to **split the code into reusable pieces**. Examples are re-using a set of lighting calculation functions and sharing a struct between two compute shaders and.
+
+However, current WGSL tooling is not built for this. The official language purposefully does not include this feature, nor does it provide adjacent features like namespaces, and there is no standardized extension yet.
+
+We also should account for **importing shader from libraries**. Ideally, users could upload WGSL shaders to existing package managers, which other users could then consume.
+
+Finally, we want **multiple tools** which can compile WGSL-with-imports down to raw WGSL. Using WGSL-with-imports both in Rust projects and in web projects should be possible.
+
+## Guide-level explanation
+
+The `import` statement extension is designed to appear somewhat familiar to TypeScript syntax,
+but with a Rust like recursive grammar to make importing
+multiple importable items less verbose.
+The syntax is also heavily inspired by [Gleam](https://gleam.run/).
+
+By placing an import at the very top of a file, one can either import an entire module, or only specific importable items, such as functions, structs or types.
+
+```
+// Importing a single item using a relative path
+import ./lighting/pbr;
+
+// Importing multiple items
+import my/geom/sphere/{ draw, default_radius as foobar };
+
+// Imports a whole module. Use it with `bevy_ui.name`
```

Comment:
This seems semantically a bit odd. Wouldn't it make more sense to have this syntax just bring in name and the other symbols into the current scope?

Reply:
It pretty much just mirrors the JavaScript star imports. 
`import * as bar from "foo/bar.wgsl";`
Would directly be translated to
`import foo/bar* as bar;`
Since that uselessly repeats `bar`, the shorthand is `import foo/bar/*;`

As for why
- We couldn't do `import foo/bar` for importing modules, because of the ambiguity issues.
- Rust-style star imports require shadowing.
- So we copied whatever Typescript does, plus a shorthand.

Reply:
Could we not disambiguate item imports by making them have to be surrounded by braces?

Reply:
We could ! I actually tried it out, and was unsure about it.
So one of the examples (which is based on code in one of the Bevy shaders) would look like this
```
import bevy_pbr/{
  forward_io/{VertexOutput},
  pbr_types/{PbrInput, pbr_input_new}
};
import bevy_pbr/pbr_bindings; // needed to be moved out of the statement above
```

Reply:
I like that more. Would also bring back the possibility of importing all symbols into the current module without having to do `module_alias.item`

Reply:
Actually, back whe I tried it out, I was using a more Rust-like syntax. So
```
import bevy_pbr::{
  forward_io::{VertexOutput},
  pbr_types::{PbrInput, pbr_input_new}
};
import bevy_pbr::pbr_bindings;
```

That kept tripping me up, because I kept thinking "ah, Rust style imports". So I always forgot that the curly braces are not optional.

At least in this variation, there's no risk of mistaking the syntax for Rust.

Reply:
adding TBD item